R/wavelets.R

```diff
@@ -27,12 +27,14 @@
 #' @importFrom torch torch_exp
 #' @param w0 the non-dimensional frequency constant. If this is set too low,
 #' then the wavelet does not sample very well: a value higher than 5 should be ok.
-#' Terrence and Compo set it to 6.
+#' Torrence and Compo set it to 6.
 #'
 #' @export
```

Comment:
You could have a Wavelet abstract class that all wavelets inherit from. It could be as simple as:


```
Wavelet <- R6::R6Class(
  "Wavelet",
  public = list(
    initialize = function() {
      
    }
  )
)
```

or slightly more complex: 

```
Wavelet <- R6::R6Class(
  "Wavelet",
  public = list(
    initialize = function() {

    },
    time = function(t, s = 1, ...) {
      stop("Not implemented")
    },
    frequency = function(w, s = 1) {
      stop("Not implemented")
    }
  )
)
```

indicating the methods that are required for a `Wavelet` kind of object.

With this you would gain:

1. A common class for all Wavelets objects, that you can check when using them.
2. Nicer error messages when Wavelets are wrongly implemented.

Reply:
Thanks for the suggestion, will do!

Reply:
Could you please take a quick look at what I did? Right now, I won't add any actual checks - let's see how the package develops, first - but it's great to have the opportunity.
Honestly, I didn't expect I'd have to duplicate any documentation in the subclasses, but after I just didn't get  `@inherit` or `@inheritParams` to work, I found this:

https://github.com/r-lib/roxygen2/issues/996

... so really it seems there is no way ... unless I'm getting something wrong ...

Reply:
Yeah, it seems you really can't inherit stuff. In theory you could use something like `@includeRmd` to avoid the repetition. There are a few examples like this in `luz`.

Reply:
What you did looks great!